keras_cv/layers/preprocessing/random_brightness.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+_SCALE_VALIDATION_ERROR = ('The `scale` should be number or a list of two numbers '
+                           'that ranged between [-1.0, 1.0]. ')
+
+
+class RandomBrightness(tf.keras.layers.Layer):
+  """Randomly adjust brightness for the a RGB image.
+
+  This layer will randomly increase/reduce the brightness for the input RGB image.
+  During inference time, the output will be identical to input. Call the layer with
+  training=True to adjust brightness of the input.
+
+  Note that same brightness adjustment will be apply to all the images in the same
```

Comment:
Is this what we want?  Does this have any positive/negative significant impact on convergence?
https://github.com/tensorflow/models/blob/master/official/vision/beta/projects/simclr/dataloaders/preprocess_ops.py#L30

Reply:
great link - yeah, if possible we should avoid applying the same augmentations to an entire batch unless it is extremely difficult to avoid doing so.

Reply:
Also if it is the same on the entire batch why we are not just layer wrapping:
https://www.tensorflow.org/api_docs/python/tf/image/random_brightness

Reply:
Also as we have discussed in https://github.com/keras-team/keras-cv/pull/91#issuecomment-1027846627 we still need to evaluate if we want to expose the op in the public API, keep it only in the keras-cv private API or just embedded in the layer itself.

In the case we want to expose the op in the public API are we going to almost duplicate `tf.image.random_brightness`?

Reply:
Updated. The new implementation will generate different adjustment for each of the image in the batch.

On a side note, tf.image.random_brightness has some assumption about the dtype and the range of the value. Eg when dtype is int, it expect the value to range from [0, 255]. When the dtype is float, it will assume the value is ranged [0.0, 1.0]. This means if you give it a float32 image with value [0, 255], then it will almost adjust nothing, since the delta is only [0, 1] and not scaled with * 255. 

The current implementation in this PR is only expecting RGB colored image, and all the dtypes should work.

Reply:
I think the difference between the current PR and tf ops is quite large and have different assumption

1. Scale the delta based on the input dtype (this is the major one, and has been apply to all the tf.image ops, which I don't think can be easily changed)
2. Support of batch inputs. All the existing tf.image ops only take care of single image input. Updating all of them to have batch input support might require more work (but doable).
3. Output value clip. (Minor and doable). 
4. Uneven delta range. (Minor and doable). 

I am not sure who is the active API owner of tf.image ops, @fchollet for more inputs.

Reply:
2. [Are you sure](https://github.com/tensorflow/tensorflow/blob/v2.8.0/tensorflow/python/ops/image_ops_impl.py#L1958)?

In the end probably it is better to take the ownership of `tf.image` here and handle the deprecation warning PRs in TF instead of growing incompatible API between TFA/TF/Keras-*.

Then I understand the everyone care only about its ownership but I think that we could improve the library perception to the users/contributors if we think as an ecosystem.

Reply:
Oh, you are right, my mistake. The current PR apply different adjustment to each image, but the tf.image apply the same adjustment.

Checked with @fchollet  offline. It seems that there isn't an active owner for tf.image at this moment. We will have some discussion about how we are going to move forward.